### ISSUE TITLE
Danger when try to change or add a new item

### ISSUE THREAD

**Issue by edelciomolina** [jcbvm/ember-i18n-editor#7]
Two issues here:
- When I try to rename a tag name that is a node in treeview, will replace all inside elements into JSON. 
- If I try to add a new item the same thing happens.

Will be more secure if we cannot to change or add new itens with the same same of the node.

**Comment by jcbvm** [comment]
Thanks for reporting. Which version are you using? The latest version should merge the trees when renaming a tag to an already existing node (exactly the same nodes will then be replaced by the one you are renaming). 

Adding a new node with a dialog will indeed replace the existing, it seems like a bug to me, should have the same functionality as the field at the left bottom where pressing enter to an existing key will not recreate it, but just selecting the existing one.

**Comment by edelciomolina** [comment]
I´m using v0.7.0. Thank you for your help, this tool is very useful and is helping us alot!

**Comment by jcbvm** [comment]
Hmm, 0.7.0 should be merge the trees just fine. I'm not able to reproduce it locally, do you have any reproduction steps (maybe a simple tranlations example) where this happens?

About the other issue, I have a fix ready for this which will be in the next release as soon as the above issue gets fixed.

**Comment by edelciomolina** [comment]
Hi @jcbvm 
In the first example I changed the name to a same name used as a node. I think that here should have a question if I wish to replace them. That would help when you is editing a lot of tags and forget that you already have one with the same name.

![test1](https://cloud.githubusercontent.com/assets/4604923/19050013/119e8760-8983-11e6-8fa5-0dc0828dbb9c.gif)

On the second example is the same problem, but now I tried to Add a new item and they disappeared. The same case of show a question here or just block this function will resolve. So, the user will see and will delete it before the add a new.

![test2](https://cloud.githubusercontent.com/assets/4604923/19050014/11a09776-8983-11e6-92cd-a7ecbab331ec.gif)

Below is my test folder with the JSON files
[locales.zip](https://github.com/jcbvm/ember-i18n-editor/files/506981/locales.zip)

Thank you!

**Comment by jcbvm** [comment]
Thanks for your great response.

I see whats happening here, I thought that this was going about renaming a tree to an exiting tree, but here you are renaming a value (single node) to a tree, in which case the result is what is expected, because the single node will replace the whole tree (there is no way we can merge here, otherwise we get invalid JSON). I think it is indeed a good idea to add a confirmation here.

The second issue is actually an issue with the tree (which I got fixed already).

Will make a fix for the problem mentioned above.
